lib/pure/collections/sequtils.nim

```diff
@@ -401,25 +401,30 @@ template foldr*(sequence, operation: expr): expr =
     result = operation
   result
 
-template mapIt*(seq1, typ, op: expr): expr =
+template mapIt*[T,S](seq1: openArray[T], op: expr): seq[S] =
   ## Convenience template around the ``map`` proc to reduce typing.
   ##
   ## The template injects the ``it`` variable which you can use directly in an
-  ## expression. You also need to pass as `typ` the type of the expression,
-  ## since the new returned sequence can have a different type than the
-  ## original.  Example:
+  ## expression. Example:
   ##
   ## .. code-block::
   ##   let
   ##     nums = @[1, 2, 3, 4]
-  ##     strings = nums.mapIt(string, $(4 * it))
+  ##     strings = nums.mapIt($(4 * it))
   ##   assert strings == @["4", "8", "12", "16"]
-  var result {.gensym.}: seq[typ] = @[]
+  type T {.gensym.} = type(seq1[0])
+  type S {.gensym.} = type((
+    block:
```

Comment:
Thanks. If you like this, then please take a look at mapIt and mapKV at https://github.com/petermora/nimLazy , I use a macro+template combo to generate mapIt.